bug_1292234.patch

```diff
@@ -128,74 +128,80 @@ diff --git a/devtools/server/actors/storage.js b/devtools/server/actors/storage.
 +    }
 +  },
 +
-+  async populateStoreMap(host, storagePrincipal, storeMap) {
-+    const db = await ExtensionStorageIDB.open(storagePrincipal);
-+    this.dbConnectionForHost.set(host, db);
-+    const data = await db.get();
++  async getStoragePrincipal(extension) {
++    if (extension.getSharedData("storageIDBBackend")) {
++      return extension.getSharedData("storageIDBPrincipal");
++    }
++    // This extension either isn't using the IDB backend (false) or
++    // it hasn't migrated to the IDB backend yet (null)
 +
-+    for (const [key, value] of Object.entries(data)) {
-+      storeMap.set(key, value);
++    // Loop for web extension context related to the current window
++    let currentContext;
++    for (const view of extension.views) {
++      currentContext = view;
++    }
++    if (!currentContext) {
++      // We didn't find the extension context for the current target; give up.
++      return null;
 +    }
-+    return storeMap;
++    const {
++      backendEnabled,
++      storagePrincipal,
++    } = await ExtensionStorageIDB.selectBackend(currentContext);
++
++    if (!backendEnabled) {
++      // IDB backend disabled; give up.
++      return null;
```

Comment:
When would `backendEnabled` return `false`? (e.g. if we're using the JSON storage backend) Do we want to throw an error here? Or a warning?

Reply:
`backendEnabled` can be false if:

- the extension failed to migrate its data to the storage.local IDB backend

or
 
- the storage.local IDB backend has been globally disabled

I think that in this case it may be nice (and reasonable) if we could show a message to the user in the storage panel itself, but I didn't looked if it is actually possible using some of the existing features of the storage actors / storage panel client.

Reply:
Okay thanks for explaining; I filed a follow-up issue #10 asking a few more clarifying questions for you.